dashboard/app/models/level.rb

```diff
@@ -349,6 +351,27 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  def log_changes
+    log = JSON.parse(audit_log || "[]")
+
+    # serialize 'properties' changes
+    changed = self.changed
+    if changed.include?('properties')
+      changed.delete('properties')
```

Comment:
I'm fairly certain that `self.changed` is a method returning a newly-constructed array each time, so I don't think that mutation is an issue, but I suppose for clarity I should isolate that operation.

As I noted above, we're deleting properties because we're exploding its entry out into entries for all of the keys of the properties blob that changed. I'll add a comment to this section explaining.